### PR TITLE
[INLONG-11252][SDK] Transform support PB sink data

### LINKED ISSUE #11252: [Feature][SDK] Transform support PB sink data
### Description

* parent issue: #10796 
* Supports pb sink encoder, provide ability to sink from pb

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by emptyOVO**
Fixes #11252 

### Motivation
add PbSinkEncoder to sink data to pb formatted byte[] , add PbSinkInfo to generate Info , provide unit tests from different data formatted Decoder

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/PbSinkEncoder.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            fieldTypes.put(field.getName(), Descriptors.FieldDescriptor.Type.STRING);
+        }
+        // decode protoDescription
+        this.dynamicDescriptor = decodeProtoDescription(pbSinkInfo.getProtoDescription());
+    }
+
+    @Override
+    public byte[] encode(SinkData sinkData, Context context) {
+        try {
+            DynamicMessage.Builder dynamicBuilder = DynamicMessage.newBuilder(dynamicDescriptor);
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/PbSinkEncoder.java:55]: Output in ProtoBuf format, it is necessary to support the scenario where multiple pieces of data are merged together for encoding. There is no application scenario for encoding a single piece of data, right?

Comment (emptyOVO) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/PbSinkEncoder.java:55]: I think of SinkData itself as a collection of data which has been transformed, and the output of transform is the multiple pieces of data that are merged together for encoding , if we need to encode them as pb we need to specify his description first, is my understanding correct?

Comment (luchunliang) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/PbSinkEncoder.java:55]: The other problem is how to encode the package header and summary fields.

Comment (luchunliang) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/encode/PbSinkEncoder.java:55]: We can improve the sink type.

**Review by luchunliang — DISMISSED** [review]

**Review by aloyszhang — APPROVED** [review]

**Comment by aloyszhang** [comment]
@emptyOVO please resolve the confict

**Comment by emptyOVO** [comment]
> @emptyOVO please resolve the confict

done

**Review by aloyszhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Pull request merged**